src/query/storage/converter_test.go

```diff
@@ -323,58 +313,163 @@ func TestPromTimeSeriesToSeriesAttributesPromMetricsType(t *testing.T) {
+			})
+
+			require.NoError(t, err)
+			assert.Equal(t, expected.metricType, attrs.PromType)
+			assert.Equal(t, expected.handleValueResets, attrs.HandleValueResets)
+			assert.Equal(t, ts.SourceTypePrometheus, attrs.Source)
+		})
 	}
 
-	_, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{Type: -1})
+	_, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+		Source: prompb.Source_PROMETHEUS,
+		Type:   -1,
+	})
 	require.Error(t, err)
 }
 
-func TestPromTimeSeriesToSeriesAttributesM3Type(t *testing.T) {
+func TestPromTimeSeriesToSeriesAttributesM3TypeFromPrometheus(t *testing.T) {
+	testPromTimeSeriesToSeriesAttributesM3Type(t, prompb.Source_PROMETHEUS)
+}
+
+func TestPromTimeSeriesToSeriesAttributesM3TypeFromOpenMetrics(t *testing.T) {
+	testPromTimeSeriesToSeriesAttributesM3Type(t, prompb.Source_OPEN_METRICS)
+}
+
+func testPromTimeSeriesToSeriesAttributesM3Type(t *testing.T, source prompb.Source) {
 	mapping := map[prompb.M3Type]ts.M3MetricType{
 		prompb.M3Type_M3_GAUGE:   ts.M3MetricTypeGauge,
 		prompb.M3Type_M3_COUNTER: ts.M3MetricTypeCounter,
 		prompb.M3Type_M3_TIMER:   ts.M3MetricTypeTimer,
 	}
 
 	for proto, expected := range mapping {
-		attrs, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{M3Type: proto})
+		attrs, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+			Source: source,
+			M3Type: proto,
+		})
 		require.NoError(t, err)
 		assert.Equal(t, expected, attrs.M3Type)
 	}
 
-	_, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{M3Type: -1})
+	_, err := PromTimeSeriesToSeriesAttributes(prompb.TimeSeries{
+		Source: source,
+		M3Type: -1,
+	})
 	require.Error(t, err)
 }
 
-func TestPromTimeSeriesToSeriesAttributesPromMetricsTypeFromGraphite(t *testing.T) {
-	mapping := map[prompb.M3Type]ts.PromMetricType{
-		prompb.M3Type_M3_GAUGE:   ts.PromMetricTypeGauge,
-		prompb.M3Type_M3_COUNTER: ts.PromMetricTypeCounter,
-		prompb.M3Type_M3_TIMER:   ts.PromMetricTypeUnknown,
+func TestPromTimeSeriesToSeriesAttributesMetricsTypeFromOpenMetrics(t *testing.T) {
+	mapping := map[prompbMetricTypeWithNameSuffix]promMetricTypeWithBool{
+		{metricType: prompb.MetricType_UNKNOWN}: {metricType: ts.PromMetricTypeUnknown},
+
+		{prompb.MetricType_COUNTER, "total"}:   {metricType: ts.PromMetricTypeCounter, handleValueResets: true},
+		{prompb.MetricType_COUNTER, "created"}: {metricType: ts.PromMetricTypeCounter},
+
+		{metricType: prompb.MetricType_GAUGE}:    {metricType: ts.PromMetricTypeGauge},
+		{metricType: prompb.MetricType_INFO}:     {metricType: ts.PromMetricTypeInfo},
+		{metricType: prompb.MetricType_STATESET}: {metricType: ts.PromMetricTypeStateSet},
+
+		{prompb.MetricType_HISTOGRAM, "bucket"}:  {metricType: ts.PromMetricTypeHistogram, handleValueResets: true},
+		{prompb.MetricType_HISTOGRAM, "count"}:   {metricType: ts.PromMetricTypeHistogram, handleValueResets: true},
+		{prompb.MetricType_HISTOGRAM, "sum"}:     {metricType: ts.PromMetricTypeHistogram, handleValueResets: true},
```

Comment:
`sum` is an interesting one. It is a sum of values registered. Values could be positive/negative. So it really depends on the domain of the value. I wonder how it's used in practice. Do people usually apply rate on `sum`?

Reply:
There should be no negative values. See https://github.com/OpenObservability/OpenMetrics/blob/2bd6413e040513afa1bfb8f56dd12db8262e18d9/specification/OpenMetrics.md#histogram

Reply:
Hmm, but with Prometheus it's possible https://prometheus.io/docs/practices/histograms/#count-and-sum-of-observations

Imagine you were to measure temperature in Celsius.

Reply:
It is possible but not advised. There is a suggestion to handle positive and negative observations separately in the doc.
So I would expect that majority of use cases will have a `rate` function. And for `rate` function to work, we must handle value resets when downsampling.